mm2src/coins/lightning/ln_events.rs

```diff
@@ -323,10 +342,38 @@ impl LightningEventHandler {
         let payment_preimage = match purpose {
             PaymentPurpose::InvoicePayment { payment_preimage, .. } => match payment_preimage {
                 Some(preimage) => *preimage,
+                // This is a swap related payment since we don't have the preimage yet
                 None => {
-                    // Free the htlc immediately if we don't have the preimage required to claim the payment
-                    // to allow for this inbound liquidity to be used for other inbound payments.
-                    self.channel_manager.fail_htlc_backwards(payment_hash);
+                    let payment_info = PaymentInfo {
```

Comment:
Is it generally safe to not call `fail_htlc_backwards` here anymore? What will happen if neither `fail_htlc_backwards` nor `claim_funds` is called from the API code afterward?

Reply:
> Is it generally safe to not call fail_htlc_backwards here anymore?

Yes, I should call it inside the swap methods implementations when it's required. I called it here previously because before swaps we shouldn't have accepted payments that we don't know the preimage for.

> What will happen if neither fail_htlc_backwards nor claim_funds is called from the API code afterward?

The `HTLC` will remain unsettled until the timelock expires in both cases.